src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -665,145 +665,169 @@ void GNAPlugin::AddDebugProperties(const InferenceEngine::CNNLayerPtr layer,
 }
 #endif
 
+static void TransformModel(std::shared_ptr<ngraph::Function> model,
```

Comment:
> What if `TransformModel` function detects that the model is not valid? eg some existing of future transform detects such a case and throws... I guess currently we silently assume that this should not happen. IMO we should not assume that at this point.

What do you mean? The idea of `TransformModel` modify graph to keep it valid. If some transformation don't do it and we have exception, we should fix it and don't hide.

Reply:
I was not suggesting hiding something.

I mean that we can have a pass which detects that the graph is not valid (valid==GNA-compliant).
and it would be good to stop loading the network at such a point (eg by thowing an exception).

If we can not throw from any n-graph transform we should agree and document this assumption.

Take a look at places where we throw from transform:
Code	File	Line	Column
        THROW_GNA_EXCEPTION << "Unsupported MVN shape size: " << mvn_shape_size;	C:\WD\repo\github\ov_intel_dns_mtl\src\plugins\intel_gna\src\transformations\decompose_mvn.cpp	77	9
                THROW_GNA_EXCEPTION << "The number of dimensions that are greater than 1 is greater than 2"	C:\WD\repo\github\ov_intel_dns_mtl\src\plugins\intel_gna\src\transformations\insert_reshape_around_matmul.cpp	101	17
            THROW_GNA_EXCEPTION << "Unable to insert transpose after: " << node->get_friendly_name()	C:\WD\repo\github\ov_intel_dns_mtl\src\plugins\intel_gna\src\transformations\insert_transpose_after_convolution_or_pooling.cpp	190	13
            THROW_GNA_EXCEPTION << "Unable to insert transpose after: " << node->get_friendly_name()	C:\WD\repo\github\ov_intel_dns_mtl\src\plugins\intel_gna\src\transformations\insert_transpose_after_convolution_or_pooling.cpp	196	13

Reply:
@kbruniec 
If graph is invalid after some transformation:
1.  From ngraph point of view exception will be raised during node validation
2.  From GNA perspective it doesn't mean anything. Because all transformations are independent. Only final graph agter all passes makes sense.